engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -396,4 +408,24 @@ private DirectBuffer getNumberOfActiveInstancesVariable(final long elementInstan
         numberOfActiveInstancesVariableView,
         numberOfActiveInstances);
   }
+
+  private DirectBuffer getNumberOfCompletedInstancesVariable(final long elementInstanceKey) {
+    final int numberOfCompletedInstances =
+        stateBehavior.getElementInstance(elementInstanceKey).getNumberOfCompletedElementInstances();
```

Comment:
🔧 You'll need to add 1 to this value when the `numberOfCompletedElementInstances` is incremented in the ProcessInstanceElementCompletedApplier instead of the ProcessInstanceElementCompletingApplier.